api/v1alpha1/watermarkpodautoscaler_types.go

```diff
@@ -204,3 +205,30 @@ type WatermarkPodAutoscalerList struct {
 func init() {
 	SchemeBuilder.Register(&WatermarkPodAutoscaler{}, &WatermarkPodAutoscalerList{})
 }
+
+// GetLogAttrs takes a list of log attrs that would be sent to logger.info, and adds user specified key value attribute
+// pairs from the ad.datadoghq.com/attributes annotation of the WPA spec.
+// logAttrs should be pairs of key/values, with each key being a string.
+// This func returns a slice! To pass this as variadic parameters to logger.info, you need the ellipses, so use wpa.GetLogAttrs()...
+func (wpa *WatermarkPodAutoscaler) GetLogAttrs(logAttrs ...interface{}) []interface{} {
+	var logAttributes []interface{}
+
+	for _, v := range logAttrs {
+		logAttributes = append(logAttributes, v)
+	}
+
+	customAttrsStr := wpa.ObjectMeta.Annotations["ad.datadoghq.com/attributes"]
```

Comment:
I would prefer to use another key for the annotation because `ad.datadoghq.com` is managed by the agent and it apply for autodiscovery features, which is not the case here. 
So I think you could use `wpa.datadoghq.com/logs-attributes` instead.

Reply:
also could you define a `const string` to store this annotation key.